.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -0,0 +1,112 @@
+name: Build and Test libc++
```

Comment:
I would like us to add a comment that explains a few aspects of this setup, because some of these choices are not arbitrary. Something like this:

```
# This file defines pre-commit CI for libc++ [bla bla bla].
#
# We split the configurations in multiple stages with the intent of saving compute time
# when a job fails early in the pipeline. This is why the jobs are marked as `continue-on-error: false`.
# We try to run the CI configurations with the most signal in the first stage.
# 
# TODO: Explain the fail-fast behavior (enabled for the first stage but not the other ones).
```